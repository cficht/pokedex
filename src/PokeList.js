import React, { Component } from 'react';
import './App.css';
import PokeItem from './PokeItem.js'
import shortid from 'shortid';
import Search from './Search';

export default class PokeList extends Component {
    render() {
        const pokeNode = this.props.pokedex.map(pokemon => <PokeItem pokemon={pokemon} key={shortid.generate()}></PokeItem>)
        return (
            <div id="list-div">
                <div id="list-top">
                    <Search />
                </div>
                <div id="list-bottom">
                    <ul>
                        {pokeNode}
                    </ul>
                </div>
            </div>
        );
    }
}