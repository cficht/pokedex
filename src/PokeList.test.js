import React from 'react';
import PokeList from './PokeList';
import renderer from "react-test-renderer";

test('renders PokeList.js correctly', () => {
  const tree = renderer
  .create(<PokeList pokedex={[]} />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});