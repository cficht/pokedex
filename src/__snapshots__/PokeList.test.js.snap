// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PokeList.js correctly 1`] = `
<div
  id="list-div"
>
  <div
    id="list-top"
  />
  <div
    id="list-bottom"
  >
    <ul />
  </div>
</div>
`;
